crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -1819,6 +1875,26 @@ impl TransportRpcModuleConfig {
         self
     }
 
+    /// Takes the http transport configuration, leaving `None` in its place.
+    pub fn take_http(&mut self) -> Option<RpcModuleSelection> {
+        self.http.take()
+    }
+
+    /// Takes the ws transport configuration, leaving `None` in its place.
+    pub fn take_ws(&mut self) -> Option<RpcModuleSelection> {
+        self.ws.take()
+    }
+
+    /// Takes the ipc transport configuration, leaving `None` in its place.
+    pub fn take_ipc(&mut self) -> Option<RpcModuleSelection> {
+        self.ipc.take()
+    }
+
+    /// Takes the module configuration, leaving `None` in its place.
+    pub fn take_config(&mut self) -> Option<RpcModuleConfig> {
+        self.config.take()
+    }
```

Comment:
can we replace these with fn (http|ws|ip)_mut -> Option<&mut>

Reply:
✅